src/views/TrackPositionAnimation/TPADecodedPositionLayer.tsx

```diff
@@ -39,8 +39,8 @@ const stylesMap = {
     'base': valuesRange.map((i) => `rgba(${baseRed}, ${baseBlue}, ${baseGreen}, ${i/5})`)
 }
 
-const draw = (context: CanvasRenderingContext2D, props: DecodeFrameProps) => {
-    const { frame, colorMap } = props
+const draw = (context: CanvasRenderingContext2D, props: DecodeFrameProps, colorMap?: ValidColorMap) => {
```

Comment:
This is that "callbacks-as-configuration" stuff I was talking about: instead of having the color map passed to the draw function as part of the frame data (props) for every frame, we configure the draw function with the callback hook in l. 64-66.